app/services/coin_market_cap_pro/update_market_metrics_service.rb

```diff
@@ -0,0 +1,48 @@
+module CoinMarketCapPro
+  class UpdateMarketMetricsService < Patterns::Service
+    include CoinMarketCapProHelpers
+
+    def call
+      update_metrics
+    end
+
+    private
+
+    def update_metrics
+      data = load_cmc_data
+      perform_update_metrics(data) if data.present?
+    end
+
+    def perform_update_metrics(data)
+      metrics = data.dig('quote', 'USD')
+      begin
+        MarketMetric.create!(
+          total_market_cap: metrics["total_market_cap"],
+          total_volume_24h: metrics["total_volume_24h"],
+          timestamp: metrics["last_updated"],
+        )
+      rescue StandardError => e
```

Comment:
No need to change but the Rubyesque thing is to just do a `rescue => error` since `StandardError` is the default.